### PR TITLE
Feat/add crdc adapters

### PR DESCRIPTION
**Pull request by nss10**
Jira Ticket: [BRH-326](https://ctds-planx.atlassian.net/browse/BRH-326)

### New Features
* Introduced 3 new Data Adapters for GDC, PDC and CIDC

### Improvements
* Eliminated a few warnings related invalid operator `|` between `'dict' | 'dict'` --> `'dict'  or  'dict'`

[BRH-326]: https://ctds-planx.atlassian.net/browse/BRH-326?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by tianj7 — APPROVED** [review]
Looking good to me, may want Craig to take a cursory look to make sure everything looks fine.

File: tests/test_agg_mds_adapters.py [hunk 1]
```diff
@@ -5654,3 +5654,1201 @@ def test_get_metadata_icdc():
             },
         },
     }
+
+
+@respx.mock
+def test_get_metadata_gdc():
```
Comment (paulineribeyre) [inline tests/test_agg_mds_adapters.py:5660]: this test file is huge, can we split these tests into one file per adapter?

**Review by craigrbarnes — APPROVED** [review]

**Review by craigrbarnes — COMMENTED** [review]
Noticed failing tests that should be addressed.

File: tests/test_agg_mds_adapters.py (hunk 1 above)
Comment (nss10) [reply tests/test_agg_mds_adapters.py]: I did split them into multiple files.

Comment (paulineribeyre) [reply tests/test_agg_mds_adapters.py]: thank you!

Comment (paulineribeyre) [reply tests/test_agg_mds_adapters.py]: I'll let Jian and Craig review the actual code changes 🙂

**Pull request merged**